src/test/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/MtAmazonDynamoDbStreamsBySharedTableTest.java

```diff
@@ -214,37 +213,43 @@ void testRecords() {
             MtRecord expected4 = putTestItem(mtDynamoDb, TENANTS[1], i++);
 
             // get shard iterator
+            CountingAmazonDynamoDbStreams dynamoDbStreams =
+                new CountingAmazonDynamoDbStreams(AmazonDynamoDbLocal.getAmazonDynamoDbStreamsLocal());
             MtAmazonDynamoDbStreams mtDynamoDbStreams = MtAmazonDynamoDbStreams.createFromDynamo(mtDynamoDb,
-                    AmazonDynamoDbLocal.getAmazonDynamoDbStreamsLocal());
+                dynamoDbStreams);
             String iterator = getShardIterator(mtDynamoDbStreams);
 
             // test without context
             assertGetRecords(mtDynamoDbStreams, iterator, expected1, expected2, expected3, expected4);
             // test with each tenant context
-            MT_CONTEXT.withContext(TENANTS[0], () -> assertGetRecords(mtDynamoDbStreams, iterator, expected1,
-                    expected2));
-            MT_CONTEXT.withContext(TENANTS[1], () -> assertGetRecords(mtDynamoDbStreams, iterator, expected3,
-                    expected4));
+            MT_CONTEXT.withContext(TENANTS[0],
+                () -> assertGetRecords(mtDynamoDbStreams, iterator, expected1, expected2));
+            MT_CONTEXT.withContext(TENANTS[1],
+                () -> assertGetRecords(mtDynamoDbStreams, iterator, expected3, expected4));
+
+            // once to fetch records, once per getRecords (3) to find empty range
+            assertEquals(4, dynamoDbStreams.getRecordsCount);
+            assertEquals(1, dynamoDbStreams.getShardIteratorCount);
```

Comment:
Mostly reformatting. This is the actual change in this class (verifying number of cache misses).